### ISSUE TITLE
EllipticalOrbitEngine cleanup

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/my-solar-system#254]
I noticed a few things while evaluating EllipticalOrbitEngine for PhET-iO instrumentation.

**Comment by pixelzoom** [comment]
In the first commit above, `sunMassProperty` was deleted.  It was declared and initialized, but never used.

In the second commit above, `this.body` was renamed to `this.planet`, to match KeplersLawsModel, and because the doc says:

```diff
    // In the case of this screen, the body0 is the sun, and body1 is the planet
    this.sun = bodies[ 0 ];
-   this.body = bodies[ 1 ];
+   this.planet = bodies[ 1 ];
```

**Comment by pixelzoom** [comment]
In https://github.com/phetsims/keplers-laws/commit/c786627ff8c6f63075497205cb8774ad072c7471, I added 8 missing `private` keywords and 2 missing `readonly` keywords.

**Comment by pixelzoom** [comment]
In the above commits, I added missing `private` and `readonly` keywords.

**Comment by pixelzoom** [comment]
In the above commit, I added an assertion that we have exactly 2 bodies, in case someone tries to use EllipticalOrbitEngine with some other orbital system.

```diff
    // In the case of this sim, the body0 is the sun, and body1 is the planet
+   assert && assert( bodies.length === 2, 'This sim requires exactly 2 bodies.' );
    this.sun = bodies[ 0 ];
    this.planet = bodies[ 1 ];
```

**Comment by pixelzoom** [comment]
In the above commit, I removed explicit type information that is inferrable.  From PhET's [TypeScript Convention](https://github.com/phetsims/phet-info/blob/main/doc/typescript-conventions.md):

> * Avoid writing type annotations when TypeScript can infer the same type.

**Comment by pixelzoom** [comment]
In the commit above, the API for Engine was changed to use `bodies: Body[]` instead of `bodies: ObservableArray<Body>`.  There is nothing in the Engine implementations that requires the additional features provided by ObservableArray.

**Comment by pixelzoom** [comment]
In the above commit, I fixed a problem with the implementation of the `update` method.

In Engine, the signature for `update` is:

```ts
  public abstract update( bodies: Body[] ): void;
```

But in EllipticalOrbitEngine, the `bodies` parameter is omitted, because `bodies` never changes in Kepler's Laws:

```
  public override update(): void {
```

Despite the fact that TypeScript will let you get away with it, changing the method signature and ignoring parameters is a bad practice.  And it's a good practice to make assumptions explict using `assert`.  So I added the ` bodies: Body[]` parameter, and an assertion that makes the assumption explicit:

```diff
-  public override update(): void {
+  public override update( bodies: Body[] ): void {
+    assert && assert( bodies === this.bodies, 'This engine expects the set of bodies to remain constant.' );
```
